pkg/volume/csi/csi_mounter.go

```diff
@@ -319,6 +322,56 @@ func (c *csiMountMgr) podAttributes() (map[string]string, error) {
 	return attrs, nil
 }
 
+func (c *csiMountMgr) podServiceAccountTokenAttrs() (map[string]string, error) {
+	if c.plugin.serviceAccountTokenGetter == nil {
+		return nil, errors.New("ServiceAccountTokenGetter is nil")
+	}
+
+	csiDriver, err := c.plugin.csiDriverLister.Get(string(c.driverName))
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(5).Infof(log("CSIDriver %q not found, not adding service account token information", c.driverName))
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	if len(csiDriver.Spec.TokenRequests) == 0 {
+		return nil, nil
+	}
+
+	outputs := map[string]string{}
+	for _, tokenRequest := range csiDriver.Spec.TokenRequests {
```

Comment:
100ms between between republish calls. the tokens are cached in kubelet. it is using the same handler as projected tokens.

Reply:
does that mean we're calling the CSI driver every 100ms?

Reply:
no because every reconciler loop takes ~3 seconds and the reconciler loop calls republish. https://github.com/kubernetes/kubernetes/blob/8be0a29ce027ecc75004c1a4945a87d128c4f45f/pkg/kubelet/volumemanager/reconciler/reconciler.go#L80-L81